whylog_vim/gui/windows.py

```diff
@@ -0,0 +1,95 @@
+import vim
+
+from whylog_vim.consts import WindowTypes
+from whylog_vim.gui.vim_commander import VimCommander
+
+
+class WindowContext(object):
+
+    def __enter__(self):
+        VimCommander.set_modifiable()
+        return VimCommander.get_buffor()
+
+    def __exit__(self, type_, value, traceback):
+        VimCommander.set_nomodifible()
+
+
+class Window(object):
+
+    def __init__(self, name, content, modifiable, size=None):
+        content = content or ''
+        self.context = WindowContext()
+        if size is None:
+            VimCommander.open_file_at_window(name)
+        else:
+            VimCommander.split_window(name)
+            VimCommander.resize(size)
+        VimCommander.set_nowritable()
+        self.set_output(content)
+        self.set_modifiable(modifiable)
+        self.name = name
+        VimCommander.set_file_type()
+        self._set_window_id()
+
+    def set_output(self, contents):
+        with self.context as output_buffer:
+            output_buffer[:] = contents.split('\n')
+
+    def set_modifiable(self, modifiable):
+        if modifiable:
+            VimCommander.set_modifiable()
+        else:
+            VimCommander.set_nomodifible()
+
+    def get_content(self):
+        with self.context as output_buffer:
+            return output_buffer[:]
+
+    def _set_window_id(self):
+        for window_id, window in enumerate(vim.windows):
```

Comment:
Windows ids shouldn't be counting by 2 different methods! It is dangerous. 
Look at https://github.com/9livesdata/whylog-vim/pull/2/files#r61730847